app/src/main/java/org/oppia/app/story/StoryFragmentPresenter.kt

```diff
@@ -0,0 +1,66 @@
+package org.oppia.app.story
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import org.oppia.app.databinding.StoryChapterViewBinding
+import org.oppia.app.databinding.StoryFragmentBinding
+import org.oppia.app.model.ChapterPlayState
+import org.oppia.app.model.ChapterSummary
+import org.oppia.app.recyclerview.BindableAdapter
+import org.oppia.app.viewmodel.ViewModelProvider
+import javax.inject.Inject
+
+/** The controller for [StoryFragment] */
+class StoryFragmentPresenter @Inject constructor(
+  private val fragment: Fragment,
+  private val viewModelProvider: ViewModelProvider<StoryViewModel>
+) {
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?, storyId: String): View? {
+    val viewModel = getStoryViewModel()
+    viewModel.setStoryId(storyId)
+    val binding = StoryFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
+
+    viewModel.storyNameLiveData.observe(fragment, Observer<String> { storyName ->
+      (fragment.activity as? AppCompatActivity)?.supportActionBar?.title = storyName
+    })
+
+    viewModel.storyChapterLiveData.observe(fragment, Observer<List<ChapterSummary>> { chapterList ->
```

Comment:
I suggest moving this logic to its own ``LiveData`` in the view model and observing it via data binding like the other live datas (this should be possible with plural strings, see my comment below for how you can generate the text string entirely using data binding).